### PR TITLE
Checking a number has exactly four factors in C++

### PR DESCRIPTION
**Pull request by ShubhamGupta577**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES

### Description

Added the code for checking a number has exactly 4 factors or not by using the number theory under CP folder.
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#6068

### CODE REVIEW

**Comment by ShubhamGupta577** [comment]
@bhav09, @ankitaggarwal23 Please review this PR.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Can you add complexity to the code. Thanks.

**Review by ankitaggarwal23 — COMMENTED** [review]
Read the instructions mentioned on Gssoc discord yesterday. Do the changes as per that.

**Comment by ShubhamGupta577** [comment]
> Read the instructions mentioned on Gssoc discord yesterday. Do the changes as per that.

I have made the changes as per the format described on Discord. Please Review this PR.

File: C-Plus-Plus/cp/Exactly_Four_Factors.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*
+A number has exactly 4 factors if below given any condition is satisfied
+ 1.If the number is a cube of a prime number.
+ 2.If the number is a product of two distinct prime numbers.
+*/
+
+#include<iostream>
+#include<vector>
+#include<math.h>
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Exactly_Four_Factors.cpp]: Replace header with ` #include <bits/stdc++.h>`

File: C-Plus-Plus/cp/Exactly_Four_Factors.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  
+  Time complexity-O(Nlog(log(N)))
+  Space complexity-O(N)
+  
+  Input: num=109
+  Output:109 does not have exactly four factors
+
+  Time complexity-O(Nlog(log(N)))
+  Space complexity-O(N)
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Exactly_Four_Factors.cpp]: Write the outputs in the way your code is producing.
Write TC and SC only once

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by ShubhamGupta577** [comment]
@ankitaggarwal23 Please review the changes.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**